In [2]:
import requests

response = requests.get('http://api.open-notify.org/iss-now.json')
status_code = response.status_code
print(status_code)

200


Requests library in Python to access and retrieve data from API using GET Request.
Status Code: Code to indicate whether everything went okay or not, different status codes for different replies.

In [3]:
response = requests.get('http://api.open-notify.org/iss-pass')
status_code = response.status_code
print(status_code)

404


Status Code 404: Indicating the resource requested was not found on Server or the end-point was not valid. In this case, the end-point was not valid since .json was not added to the end of the end-point name.

In [4]:
response = requests.get('http://api.open-notify.org/iss-pass.json')
status_code = response.status_code
print(status_code)

400


Status Code 400: Indicating that a bad request was made. In this case, we have to pass 2 parameters, lat and lon i.e the coordinates of where we want.

In [5]:
parameters = {'lat': 40.71, 'lon': -74}
response = requests.get('http://api.open-notify.org/iss-pass.json',params=parameters)
print(response.content)

b'{\n  "message": "success", \n  "request": {\n    "altitude": 100, \n    "datetime": 1475258368, \n    "latitude": 40.71, \n    "longitude": -74.0, \n    "passes": 5\n  }, \n  "response": [\n    {\n      "duration": 116, \n      "risetime": 1475270687\n    }, \n    {\n      "duration": 611, \n      "risetime": 1475276175\n    }, \n    {\n      "duration": 622, \n      "risetime": 1475281956\n    }, \n    {\n      "duration": 553, \n      "risetime": 1475287824\n    }, \n    {\n      "duration": 567, \n      "risetime": 1475293669\n    }\n  ]\n}\n'


In [6]:
parameters = {'lat': 37.78, 'lon': -122.41}
response = requests.get('http://api.open-notify.org/iss-pass.json',params=parameters)
content = response.content
print(content)

b'{\n  "message": "success", \n  "request": {\n    "altitude": 100, \n    "datetime": 1475258421, \n    "latitude": 37.78, \n    "longitude": -122.41, \n    "passes": 5\n  }, \n  "response": [\n    {\n      "duration": 198, \n      "risetime": 1475281707\n    }, \n    {\n      "duration": 621, \n      "risetime": 1475287237\n    }, \n    {\n      "duration": 602, \n      "risetime": 1475293040\n    }, \n    {\n      "duration": 489, \n      "risetime": 1475298943\n    }, \n    {\n      "duration": 500, \n      "risetime": 1475304808\n    }\n  ]\n}\n'


Passing Latitude and Longitude parameters and retrieving the content.

In [7]:
import json

parameters = {"lat": 37.78, "lon": -122.41}
response = requests.get("http://api.open-notify.org/iss-pass.json", params=parameters)

data = response.json()
print(type(data))
print(data)

<class 'dict'>
{'response': [{'duration': 198, 'risetime': 1475281707}, {'duration': 621, 'risetime': 1475287237}, {'duration': 602, 'risetime': 1475293040}, {'duration': 489, 'risetime': 1475298943}, {'duration': 500, 'risetime': 1475304808}], 'message': 'success', 'request': {'longitude': -122.41, 'latitude': 37.78, 'datetime': 1475261523, 'altitude': 100, 'passes': 5}}


We use JSON (Java Script object Notation) to easily handle data from API Endpoints. We use JSON to do operations on data retrieved from the API.
And the type of data retrieved is Dict.

In [8]:
first_pass_duration = data['response'][0]['duration']
print(first_pass_duration)

198


The JSON data returned is a dictionary, so using the Key values we can access individual elements within the data.

In [9]:
print(response.headers)
content_type = response.headers['content-type']
print(content_type)

{'Via': '1.1 vegur', 'Date': 'Fri, 30 Sep 2016 18:52:03 GMT', 'Server': 'gunicorn/19.6.0', 'Content-Type': 'application/json', 'Content-Length': '521', 'Connection': 'keep-alive'}
application/json


The response contains the metadata i.e how the data was generated and how it is to be decoded. So, the content-type is of type .json, so we can use the JSON package to decode it.

In [10]:
response = requests.get('http://api.open-notify.org/astros.json')
data = response.json()
print(data)

in_space_count = data['number']
print(in_space_count)

{'people': [{'name': 'Anatoly Ivanishin', 'craft': 'ISS'}, {'name': 'Takuya Onishi', 'craft': 'ISS'}, {'name': 'Kate Rubins', 'craft': 'ISS'}], 'message': 'success', 'number': 3}
3


Shows the number of people currently in space and their information.